src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -2162,5 +2172,20 @@ namespace Opm {
         return this->m_actions;
     }
 
+
+    void Schedule::applyAction(size_t reportStep, const ActionX& action, const std::vector<std::string>& matching_wells) {
+        ParseContext parseContext;
+        ErrorGuard errors;
+
+        for (const auto& keyword : action) {
+            if (actionx_whitelist.find(keyword.name()) == actionx_whitelist.end())
```

Comment:
I think this might be better in a helper function to reduce the amount of low-level manipulation.

Reply:
Well; I disagree here. The function we are looking at will (most probably) be changed quite a lot in the future anyways, so I think it can be as is.

Reply:
I was talking explicitly about the `if (find() == .end())` condition.  That should be a separate helper function.  Then you could also wrap the whitelist into said function to reduce number of direct globals.